src/SignalR/server/Core/src/SignalRConnectionBuilderExtensions.cs

```diff
@@ -13,13 +14,15 @@ namespace Microsoft.AspNetCore.SignalR
     /// </summary>
     public static class SignalRConnectionBuilderExtensions
     {
+        private const DynamicallyAccessedMemberTypes HubAccessibility = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.PublicMethods;
+
         /// <summary>
         /// Configure the connection to host the specified <see cref="Hub"/> type.
         /// </summary>
         /// <typeparam name="THub">The <see cref="Hub"/> type to host on the connection.</typeparam>
         /// <param name="connectionBuilder">The connection to configure.</param>
         /// <returns>The same instance of the <see cref="IConnectionBuilder"/> for chaining.</returns>
-        public static IConnectionBuilder UseHub<THub>(this IConnectionBuilder connectionBuilder) where THub : Hub
+        public static IConnectionBuilder UseHub<[DynamicallyAccessedMembers(HubAccessibility)]THub>(this IConnectionBuilder connectionBuilder) where THub : Hub
```

Comment:
Hmm, this should be done in gRPC.

Reply:
Yes

Reply:
https://github.com/rynowak/link-a-thon/blob/master/src/GrpcService/Linker.xml